src/utils/createTabNavigator.js

```diff
@@ -144,6 +145,16 @@ export default function createTabNavigator(TabView: React.ComponentType<*>) {
       }
     };
 
+    _handleTabLongPress = ({ route }) => {
+      const { descriptors } = this.props;
+      const descriptor = descriptors[route.key];
+      const { navigation, options } = descriptor;
+
+      if (options.tabBarOnLongPress) {
+        options.tabBarOnLongPress({ navigation });
```

Comment:
I think the default behavior for long press should be to navigate to the tab if it's not focused already. We should pass in a defaultHandler to tabBarOnLongPress in the same way that tabBarOnPress works :)